mdp/nodes/basis_function_nodes.py

```diff
@@ -0,0 +1,466 @@
+import mdp
+from mdp.utils import mult
+import itertools
+from misc_nodes import GridProcessingNode
+
+
+class BasisFunctionNode(mdp.Node):
+    """
+    BasisFunctionNode is a non-trainable node that
+    provides a set of non-linear basis functions,
```

Comment:
I don't really understand what this node does based on this docstring. What does "provides" mean in this context?